app.ts

```diff
@@ -127,38 +120,99 @@ export type CompilerExplorerOptions = Partial<{
-    noCache: [Boolean],
-    // Don't cleanly run if two or more compilers have clashing ids
-    ensureNoIdClash: [Boolean],
-    logHost: [String],
-    logPort: [Number],
-    hostnameForLogging: [String],
-    suppressConsoleLog: [Boolean],
-    metricsPort: [Number],
-    loki: [String],
-    discoveryonly: [String],
-    prediscovered: [String],
-    version: [Boolean],
-    webpackContent: [String],
-    noLocal: [Boolean],
-}) as CompilerExplorerOptions;
+// Define an interface for our Commander options
+interface CommanderOptions extends OptionValues {
+    env: string[];
+    rootDir: string;
+    host?: string;
+    port?: string;
+    propDebug?: boolean;
+    debug?: boolean;
+    dist?: boolean;
+    archivedVersions?: string;
+    remoteFetch: boolean; // Note: this is inverted from our target
+    tmpDir?: string;
+    wsl?: boolean;
+    language?: string[];
+    cache: boolean; // Note: this is inverted from our target
+    ensureNoIdClash?: boolean;
+    logHost?: string;
+    logPort?: string;
+    hostnameForLogging?: string;
+    suppressConsoleLog?: boolean;
+    metricsPort?: string;
+    loki?: string;
+    discoveryOnly?: string;
+    prediscovered?: string;
+    webpackContent?: string;
+    local: boolean; // Note: this is inverted from our target
+    version?: boolean;
+}
+
+// Parse arguments from command line
+const program = new Command();
+program
+    .name('compiler-explorer')
+    .description('Interactively investigate compiler output')
+    .option('--env <environments...>', 'Environment(s) to use', ['dev'])
+    .option('--root-dir <dir>', 'Root directory for config files', './etc')
+    .option('--host <hostname>', 'Hostname to listen on')
+    .option('--port <port>', 'Port to listen on', '10240')
+    .option('--prop-debug', 'Debug properties')
+    .option('--debug', 'Enable debug output')
+    .option('--dist', 'Running in dist mode')
+    .option('--archived-versions <path>', 'Path to archived versions')
+    .option('--no-remote-fetch', 'Ignore fetch marks and assume every compiler is found locally')
+    .option('--tmp-dir <dir>', 'Directory to use for temporary files')
+    .option('--wsl', 'Running under Windows Subsystem for Linux')
+    .option('--language <languages...>', 'Only load specified languages for faster startup')
+    .option('--no-cache', 'Do not use caching for compilation results')
+    .option('--ensure-no-id-clash', "Don't run if compilers have clashing ids")
+    .option('--log-host <hostname>', 'Hostname for remote logging')
+    .option('--log-port <port>', 'Port for remote logging')
+    .option('--hostname-for-logging <hostname>', 'Hostname to use in logs')
+    .option('--suppress-console-log', 'Disable console logging')
+    .option('--metrics-port <port>', 'Port to serve metrics on')
+    .option('--loki <url>', 'URL for Loki logging')
+    .option('--discovery-only <file>', 'Output discovery info to file and exit')
```

Comment:
Does this still work or does it need to change? https://github.com/compiler-explorer/infra/blob/main/init/do-discovery.sh#L22

Reply:
Good catch: I'll update the other side, this is not an intentional change but I prefer nothavingwordsrunon in favour of --unix-style-options anyway. I'll ensure this works (and/or add an alias for the old behaviour in the interim)

Reply:
Fixed, tested and reinstated nearly all the other names as synonyms at least for now.